smartsim/_core/launcher/dragon/dragonLauncher.py

```diff
@@ -318,13 +132,38 @@ def run(self, step: Step) -> t.Optional[str]:
         cmd = step.get_launch_cmd()
         out, err = step.get_output_files()
 
-        if isinstance(step, DragonStep):
+        if isinstance(step, DragonBatchStep):
+            # wait for batch step to submit successfully
+            sublauncher_step_id: t.Optional[str] = None
+            return_code, out, err = self.task_manager.start_and_wait(cmd, step.cwd)
+            if return_code != 0:
+                raise LauncherError(f"Sbatch submission failed\n {out}\n {err}")
+            if out:
+                sublauncher_step_id = out.strip()
+                logger.debug(
+                    f"Gleaned batch job id: {sublauncher_step_id} for {step.name}"
+                )
+
+            if sublauncher_step_id is None:
+                raise SmartSimError("Could not get step id for batch step")
+
+            if isinstance(step.batch_settings, SbatchSettings):
```

Comment:
This looks like it could be solved w/

`sublauncher.step_mapping.add(...)`

in place of the if/elif.

We're also now doing step_id building in multiple places and it seems like it won't be maintainable